web/src/engine/osk/src/input/gestures/browser/modipress.ts

```diff
@@ -0,0 +1,71 @@
+import { type KeyElement } from '../../../keyElement.js';
+import VisualKeyboard from '../../../visualKeyboard.js';
+
+import { KeyDistribution, ActiveKeyBase, ManagedPromise } from '@keymanapp/keyboard-processor';
+import { GestureSequence } from '@keymanapp/gesture-recognizer';
+import { GestureHandler } from '../gestureHandler.js';
+
+/**
+ *
+ */
+export default class Modipress implements GestureHandler {
+  private completionCallback: () => void;
+  private originalLayer: string;
+  private shouldRestore: boolean = false;
+  private source: GestureSequence<KeyElement, string>;
+
+  constructor(
+    source: GestureSequence<KeyElement, string>,
+    vkbd: VisualKeyboard,
+    completionCallback: () => void,
+  ) {
+    const initialStage = source.stageReports[0];
+    this.originalLayer = initialStage.sources[0].stateToken;
+    this.source = source;
+
+    this.completionCallback = () => {
+      vkbd.lockLayer(false);
+      if(this.shouldRestore) {
+        vkbd.layerId = this.originalLayer;
+        vkbd.updateState();
+      }
+      completionCallback && completionCallback();
```

Comment:
```suggestion
      completionCallback?.();
```

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#optional_chaining_with_function_calls